packages/protocol/contracts/dapis/rrpDapiServer.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "../AirnodeRrp.sol";
+import "./reducers/MeanMedianHybrid.sol";
+
+abstract contract rrpDapiServer is MeanMedianHybrid {
+    struct DapiParameters {
+        uint256 minResponsesToReduce;
+        bytes32[] templateIds;
+        address[] designatedWallets;
+        address reduceAddress;
+        bytes4 reduceFunctionId;
+        }
+    
+    AirnodeRrp public airnodeRrp;
+    mapping(bytes32 => DapiParameters) private dapiIdToParameters;
+    uint256 public requesterIndex;
+    uint256 public nextDapiRequestIndex = 1;
+    mapping(bytes32 => uint256) private requestIdToDapiRequestIndex;
+    mapping(uint256 => int256[]) private dapiRequestIndexToResponses;
+    mapping(uint256 => bytes32) private dapiRequestIndexToDapiId;
+
+    constructor(address _airnodeRrp) {
+        airnodeRrp = AirnodeRrp(_airnodeRrp);
+    }
+
+    function makeDapiRequest(bytes32 dapiId)
+        external
+    {
+        DapiParameters storage dapiParameters = dapiIdToParameters[dapiId];
+        require(dapiParameters.reduceAddress == msg.sender);
```

Comment:
So this contract is called by the dAPI/aggregator contract?

Reply:
This contract is called by a price data feed contract or a prediction market for example, so essentially like an Airnode client contract